### PR TITLE
Sixth homework

### PR DESCRIPTION
**Pull request by AlexeiVinchester**
It works good with statistics, with redux-persist only statistics slice get into localStorage. I have only one problem, and if I understand well, it is also connected with types:

Снимок экрана 2024-07-30 в 19 10 41

Снимок экрана 2024-07-30 в 19 10 59

Снимок экрана 2024-07-30 в 19 12 56

I have thoughts to add custom type for action.payload. All this code works, but when I appeal in reducer to state with it's structure on the screenshot below, I've got angry vs code))

### CODE REVIEW

File: src/components/pages/ResultQuizPage/ResultQuizPage.tsx [hunk 1]
```diff
@@ -4,42 +4,56 @@ import { QuizConfigContainer } from "./QuizConfigContainer/QuizConfigContainer";
...
+            category,
+            difficulty, 
+            type
+        }));
+    };
+
+    useEffect(() => {
+        console.log('Mounting result pade ande geting quiz results');
+        saveStatistics()
+    });
```
Comment (EugeneSolovyev) [inline src/components/pages/ResultQuizPage/ResultQuizPage.tsx]: I think this is a bad practice to call useEffect each time
please think about dependency here or about handler to save statistic

File: src/redux/slices/statisticsSlice.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { initialStatistics } from "./interfaces/statisticsSlice.interface";
+
+const statisticsSlice = createSlice({
+    name: 'statistics',
+    initialState: { ...initialStatistics },
+    reducers: {
+        addCurrentQuizResultToStatistics(state, action) {
+            console.log(action.payload)
```
Comment (EugeneSolovyev) [inline src/redux/slices/statisticsSlice.ts]: please remove console.logs

File: src/components/pages/ResultQuizPage/ResultQuizPage.tsx [hunk 3]
```diff
@@ -31,20 +31,19 @@ const ResultQuizPage = () => {
...
         }));
-    };
+    }, [correctAnswers, questions, category, difficulty, type, dispatch]);
 
     useEffect(() => {
-        console.log('Mounting result pade ande geting quiz results');
-        saveStatistics()
-    });
+        saveStatistics();
+    }, [saveStatistics]);
```
Comment (EugeneSolovyev) [inline src/components/pages/ResultQuizPage/ResultQuizPage.tsx]: I see what you want to reach
```tsx
    useEffect(() => {
dispatch(addCurrentQuizResultToStatistics({
            questions: questions.length,
            correctAnswers,
            category,
            difficulty, 
            type
        }));
    }, [correctAnswers, questions, category, difficulty, type, dispatch]));
```
this is more clear
in your case no need to create a callback :)

**Comment by AlexeiVinchester** [comment]
fixed all problems with types about what I had wrote before. And also fixed problem with dependencies.

**Review by EugeneSolovyev — APPROVED** [review]

**Pull request merged**